src/scenes/ObservationsRouter/adhoc/index.tsx

```diff
@@ -1,113 +1,170 @@
 import { useLocalization } from 'src/providers';
-import { selectObservationMonitoringPlot } from 'src/redux/features/observations/observationMonitoringPlotSelectors';
-import { selectObservationsResults } from 'src/redux/features/observations/observationsSelectors';
+import { useSpeciesData } from 'src/providers/Species/SpeciesContext';
+import { usePlantingSiteData } from 'src/providers/Tracking/PlantingSiteContext';
 import { getConditionString } from 'src/redux/features/observations/utils';
-import { selectPlantingSite } from 'src/redux/features/tracking/trackingSelectors';
-import { useAppSelector } from 'src/redux/store';
 import DetailsPage from 'src/scenes/ObservationsRouter/common/DetailsPage';
 import MonitoringPlotPhotos from 'src/scenes/ObservationsRouter/common/MonitoringPlotPhotos';
 import SpeciesMortalityRateChart from 'src/scenes/ObservationsRouter/common/SpeciesMortalityRateChart';
 import SpeciesTotalPlantsChart from 'src/scenes/ObservationsRouter/common/SpeciesTotalPlantsChart';
 import strings from 'src/strings';
+import {
+  ObservationMonitoringPlotResultsPayload,
+  ObservationPlantingSubzoneResultsPayload,
+  ObservationPlantingZoneResultsPayload,
+  ObservationSpeciesResults,
+} from 'src/types/Observations';
 import { getShortTime } from 'src/utils/dateFormatter';
 import useDeviceInfo from 'src/utils/useDeviceInfo';
 import { useDefaultTimeZone } from 'src/utils/useTimeZoneUtils';
 
 export default function ObservationMonitoringPlot(): JSX.Element | undefined {
-  const { plantingSiteId, observationId, plantingZoneName, monitoringPlotId } = useParams<{
+  const params = useParams<{
     plantingSiteId: string;
     observationId: string;
     plantingZoneName: string;
     monitoringPlotId: string;
   }>();
+
+  const plantingSiteId = Number(params.plantingSiteId);
+  const observationId = Number(params.observationId);
+  const plantingZoneName = params.plantingZoneName;
+  const monitoringPlotId = Number(params.monitoringPlotId);
+
   const defaultTimeZone = useDefaultTimeZone();
   const navigate = useSyncNavigate();
   const theme = useTheme();
   const { isMobile } = useDeviceInfo();
   const { activeLocale } = useLocalization();
-  const allObservationsResults = useAppSelector(selectObservationsResults);
-  const observationsResults = useMemo(() => {
-    if (!allObservationsResults || !plantingSiteId) {
-      return [];
+
+  const { species } = useSpeciesData();
+  const { plantingSite, adHocObservationResults, observationResults } = usePlantingSiteData();
+  const [plantingZoneResult, setPlantingZoneResult] = useState<ObservationPlantingZoneResultsPayload>();
+  const [plantingSubzoneResult, setPlantingSubzoneResult] = useState<ObservationPlantingSubzoneResultsPayload>();
+  const [monitoringPlotResult, setMonitoringPlotResult] = useState<ObservationMonitoringPlotResultsPayload>();
+
+  const result = useMemo(() => {
+    if (!Number.isNaN(Number(observationId))) {
```

Comment:
You're already converting to a Number on line 40, so this check is unnecessary, or the Number conversions above need to be removed